From b1c13f61a2012f35466768be1de0b3761934fc3a Mon Sep 17 00:00:00 2001
From: Radostin Stoyanov <radostin@redhat.com>
Date: Tue, 14 Dec 2021 12:54:19 +0000
Subject: [PATCH 069/249] Add documentation for --timeout option

The --timeout option was introduced in [1] to prevent criu dump from
being able to hang indefinitely and allow users to adjust the time limit
in seconds for collecting tasks during the dump operation.

[1] https://github.com/checkpoint-restore/criu/commit/d0ff730

Signed-off-by: Radostin Stoyanov <radostin@redhat.com>
---
 Documentation/criu.txt | 4 ++++
 criu/crtools.c         | 2 ++
 2 files changed, 6 insertions(+)

diff --git a/Documentation/criu.txt b/Documentation/criu.txt
index 4c6885fc9..f41b1898c 100644
--- a/Documentation/criu.txt
+++ b/Documentation/criu.txt
@@ -367,6 +367,10 @@ mount -t cgroup -o devices,freezer none devices,freezer
     Allows to link unlinked files back, if possible (modifies filesystem
     during *restore*).
 
+*--timeout* 'number'::
+    Set a time limit in seconds for collecting tasks during the
+    dump operation. The timeout is 10 seconds by default.
+
 *--ghost-limit* 'size'::
     Set the maximum size of deleted file to be carried inside image.
     By default, up to 1M file is allowed. Using this
diff --git a/criu/crtools.c b/criu/crtools.c
index 81c0aa963..da47bd684 100644
--- a/criu/crtools.c
+++ b/criu/crtools.c
@@ -391,6 +391,8 @@ usage:
 	       "  -j|--" OPT_SHELL_JOB "        allow one to dump and restore shell jobs\n"
 	       "  -l|--" OPT_FILE_LOCKS "       handle file locks, for safety, only used for container\n"
 	       "  -L|--libdir           path to a plugin directory (by default " CR_PLUGIN_DEFAULT ")\n"
+	       "  --timeout NUM         a timeout (in seconds) on collecting tasks during dump\n"
+	       "                        (default 10 seconds)\n"
 	       "  --force-irmap         force resolving names for inotify/fsnotify watches\n"
 	       "  --irmap-scan-path FILE\n"
 	       "                        add a path the irmap hints to scan\n"
-- 
2.35.1

